app/models/manageiq/providers/autosde/inventory/collector/target_collection.rb

```diff
@@ -18,15 +18,15 @@ def storage_resources
   end
 
   def storage_hosts
-    return [] if references(:storage_hosts).blank?
+    []
```

Comment:
These were returning `nil` instead of `[]` if there were references to these (follow-up to https://github.com/ManageIQ/manageiq-providers-autosde/pull/107#discussion_r759270735)